sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -426,23 +426,22 @@ public new AppendBlobClient WithSnapshot(string snapshot)
                     $"{nameof(Uri)}: {Uri}\n" +
                     $"{nameof(httpHeaders)}: {httpHeaders}");
                 var conditions = new AppendBlobRequestConditions { IfNoneMatch = new ETag(Constants.Wildcard) };
+                Response<BlobContentInfo> response;
                 try
                 {
-                    Response<BlobContentInfo> response = await CreateInternal(
+                    response = await CreateInternal(
                         httpHeaders,
                         metadata,
                         conditions,
                         async,
                         cancellationToken,
                         $"{nameof(AppendBlobClient)}.{nameof(CreateIfNotExists)}")
                         .ConfigureAwait(false);
-
-                    return response;
                 }
                 catch (RequestFailedException storageRequestFailedException)
                 when (storageRequestFailedException.ErrorCode == BlobErrorCode.BlobAlreadyExists)
                 {
-                    return default;
+                    response = Response.FromValue(new BlobContentInfo(), default);
```

Comment:
I worry about shipping this.  If anyone's checking for null today, they'll stop working.  Maybe there's nothing we can do here but document usage patterns.

Reply:
@tg-msft Would changing this `Response<T>` to be null-tolerant be a viable alternative?

Reply:
Note that in the doc comments for this specific method we already noted that if the operation failed, we would return `null`.  At a minimum, if we are to make this change, we need to update the doc comment for this method.

Reply:
Also - it's very odd that there is now a case where I get a `Response<T>` which wraps a default value for `BlobContentInfo`.  Previous code that was written against the documentation will assume that if a `Response<BlobContentInfo>` is returned, the information in the `BlobContentInfo` reflects the uploaded blob.  In the case where the blob already exists, that is no longer the case, instead I have what looks like a successful response but just contains default data.

Reply:
Changing `Response<T>`'s implicit cast to return `response?.Value ?? default` is tricky too.  @pakrym / @ellismg - what's your take on whether replacing a `NullRefEx` with returning `default` counts as a break per https://github.com/ltrzesniewski/dotnet-runtime/blob/master/docs/coding-guidelines/breaking-change-rules.md#exceptions?

Reply:
We would have to remove the non-nullable annotation of it which might be breaking for some code bases.

Reply:
Nullability annotations aside (I think that touches on source vs binary breaking changes), it's not clear to me that this ends up being better.  In general, it is safe to write code that looks like this today (this API not withstanding):

```csharp
SomeModelClass c = await client.SomeServiceMethodThatReturnsResposeOfSomeModelClass();
if (c.SomeProperty) {
   // kaboom!
}
```

It feels like this change would break that invariant (for anyone using Response<T>, they would have to be defensive) and that doesn't feel like the kind of change we should be making to one of our core types.  I think saying "most of the time Response<T> will have a value, but in some methods where the SDK augments service logic, it may return null" is going to lead to more confusing long term, requiring me to carefully read the documentation for every method I call where I want to use a response to understand if the result will be null or not or slather my code in null checks.

It feels like this is a rather big change to `Response<T>` from a semantics point of view, but I could be over biased there.  I think there are more tactical ways for us to fix this issue, but I think understanding what we do for these sorts of `XXXIfNotExists` methods should be across all our SDKs.  It does feel like having `Response<T>` as the return value may not be sufficient, and we need some better way to denote the case where the operation you were trying to preform did not happen because the resource already existed.